Finalize PostgreSQL configuration and cleanup H2 references.

MJCarvajalQ opened the pull request:
🎯 **Purpose**

This PR finalizes the PostgreSQL configuration and removes all H2 references from the main application, completing the transition to PostgreSQL as the primary database. H2 remains only in test configuration for fast unit testing. Closes #20

📋 **Changes Made**

- Configure PostgreSQL dialect in JPA settings (`org.hibernate.dialect.PostgreSQLDialect`)
- Add HikariCP connection pool configuration with optimized settings:
  - Maximum pool size: 20
  - Minimum idle: 5
  - Connection timeout, idle timeout, max lifetime, and leak detection thresholds
- Configure database logging for different environments:
  - Development: DEBUG/TRACE level for SQL queries and parameter binding
  - Production: WARN level to reduce log noise
- Enable HikariCP debug logging in development for connection pool monitoring
- Keep H2 database only in `application-test.properties` for unit tests

🔍 **Files Changed**

- `src/main/resources/application.properties` - Added HikariCP connection pool settings, PostgreSQL dialect
- `src/main/resources/application-dev.properties` - Added detailed database logging (DEBUG/TRACE)
- `src/main/resources/application-prod.properties` - Configured minimal logging (WARN)
- `src/test/resources/application-test.properties` - H2 configuration preserved for testing

✅ **Testing**

✅ Application starts successfully with PostgreSQL using `dev` profile  
✅ Liquibase creates schema correctly on empty database  
✅ HikariCP connection pool initializes
✅ All API endpoints work correctly with PostgreSQL (GET, POST operations)  
✅ Data persists correctly between application restarts  
✅ H2 remains functional for unit tests (`test` profile)  
✅ SQL queries and connection pool activity visible in development logs  
✅ Tested with both empty and seeded databases

📝 **Notes**

- PostgreSQL is now the primary database for all non-test environments
- H2 isolated to test scope only (`application-test.properties`)
- Development logging provides comprehensive visibility (SQL queries)

📝 Questions for Review

Transaction Management: The technical notes mention "Ensure proper transaction management." Currently, the service layer relies on Spring Data JPA's implicit transaction management for repository methods. Should I add explicit @Transactional annotations to service methods (e.g., @Transactional for write operations and @Transactional(readOnly = true) for read operations), or is the current implicit handling sufficient?

MJCarvajalQ committed:
Finalize PostgreSQL configuration and cleanup H2 references.
- Add complete HikariCP connection pool configuration
- Configure database logging for dev (DEBUG/TRACE) and prod (WARN) profiles
- Remove all H2 references from main application (H2 only in test config)
- Verify PostgreSQL as primary database with all endpoints working
Closes #20

MarcinLason reviewed (approved):
One comment into your consideration, but I have no strong opinion which version is better. Approving.

MJCarvajalQ committed:
refactor: make environment-specific config files self-contained

- Move all properties from application.properties to environment-specific files
- Each profile (dev/prod/test) now contains all required properties explicitly
- Keep application.properties minimal with only shared metadata (spring.application.name)
- Improves readability by making each environment configuration self-documenting
- Reduces ambiguity about which properties apply to which environment

Closes #20

MJCarvajalQ commented on src/main/resources/application-dev.properties:
```diff
@@ -10,4 +10,10 @@ spring.jpa.show-sql=true
+# Database logging
+logging.level.org.hibernate.SQL=DEBUG
+logging.level.org.hibernate.type.descriptor.sql.BasicBinder=TRACE
+logging.level.com.zaxxer.hikari=DEBUG
```
Also, I reviewed my service layer and confirmed that each method performs a single write operation. Since I'm not updating multiple entities in the same transaction, I'm relying on Spring's default transaction handling as you suggested. I will read more about it too :)
